packages/safe-ds-vscode/src/extension/pythonServer.ts

```diff
@@ -282,11 +282,23 @@ export const executePipeline = async function (services: SafeDsServices, pipelin
     }
     mainPipelineName = services.builtins.Annotations.getPythonName(firstPipeline) || firstPipeline.name;
     if (pipelinePath.endsWith('.sdspipe')) {
-        mainModuleName = path.basename(pipelinePath, '.sdspipe').replaceAll('-', '_');
+        mainModuleName = path
+            .basename(pipelinePath, '.sdspipe')
+            .replaceAll('%2520', '_')
+            .replaceAll(/[ .-]/gu, '_')
+            .replaceAll(/\\W/gu, '');
```

Comment:
Extract this into a function.

Reply:
As this was copied from the code generator, I'll add a helper function to the python generator that can be called here